experimental/casemap/src/titlecase.rs

```diff
@@ -0,0 +1,401 @@
+/// assert_eq!(cm_legacy.titlecase_segment_to_string("49ers", &root, default_options), "49Ers");
+/// assert_eq!(cm_normal.titlecase_segment_to_string("49ers", &root, no_adjust), "49ers");
+/// ```
+/// <div class="stab unstable">
+/// 🚧 This code is experimental; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. It can be enabled with the "experimental" Cargo feature
+/// of the icu meta-crate. Use with caution.
+/// <a href="https://github.com/unicode-org/icu4x/issues/2535">#2535</a>
+/// </div>
+#[derive(Clone, Debug)]
+pub struct TitlecaseMapper<CM> {
+    cm: CM,
+    gc: Option<DataPayload<GeneralCategoryV1Marker>>,
+}
+
+impl TitlecaseMapper<CaseMapper> {
+    /// A constructor which creates a [`TitlecaseMapper`], with the normal (non-legacy) head adjustment behavior.
+    /// See struct docs on [`TitlecaseMapper`] for more information on head adjustment behavior and usage examples.
+    ///
+    /// ✨ *Enabled with the `compiled_data` Cargo feature.*
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    #[cfg(feature = "compiled_data")]
+    pub const fn new() -> Self {
+        Self {
+            cm: CaseMapper::new(),
+            gc: Some(DataPayload::from_static_ref(
+                icu_properties::provider::Baked::SINGLETON_PROPS_GC_V1,
+            )),
+        }
+    }
+    /// A constructor which creates a [`TitlecaseMapper`], with the legacy head adjustment behavior.
+    /// See struct docs on [`TitlecaseMapper`] for more information on head adjustment behavior and usage examples.
+    ///
+    /// ✨ *Enabled with the `compiled_data` Cargo feature.*
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    #[cfg(feature = "compiled_data")]
+    pub const fn new_legacy() -> Self {
+        Self {
+            cm: CaseMapper::new(),
+            gc: None,
+        }
+    }
+
+    icu_provider::gen_any_buffer_data_constructors!(locale: skip, options: skip, error: DataError,
+    #[cfg(skip)]
+    functions: [
+        new,
+        try_new_with_any_provider,
+        try_new_with_buffer_provider,
+        try_new_unstable,
+        Self,
+    ]);
+    icu_provider::gen_any_buffer_data_constructors!(locale: skip, options: skip, error: DataError,
+    #[cfg(skip)]
+    functions: [
+        new_legacy,
+        try_new_legacy_with_any_provider,
+        try_new_legacy_with_buffer_provider,
+        try_new_legacy_unstable,
+        Self,
+    ]);
+
+    #[doc = icu_provider::gen_any_buffer_unstable_docs!(UNSTABLE, Self::new)]
+    pub fn try_new_unstable<P>(provider: &P) -> Result<Self, DataError>
+    where
+        P: DataProvider<CaseMapV1Marker> + DataProvider<GeneralCategoryV1Marker> + ?Sized,
+    {
+        let cm = CaseMapper::try_new_unstable(provider)?;
+        let gc = Some(provider.load(Default::default())?.take_payload()?);
```

Comment:
And this should be `Some(icu_properties::maps::load_general_category(provider)?)`